scrc/dataset_creation/dataset_creator.py

```diff
@@ -266,20 +270,100 @@ def get_dataset_folder(self):
             return self.create_dir(self.tmp_subdir, self.dataset_name)
         return self.create_dir(self.datasets_subdir, self.dataset_name)
 
-    def create_dataset(self, sub_datasets=False, kaggle=False, save_reports=False):
+    def create_dataset(self, court_list=None, concatenate=False, sub_datasets=False, kaggle=False, save_reports=False):
         """
         Retrieves the respective function named by the dataset and executes it to get the df for that dataset.
         :return:
         """
+        if court_list is None:
+            court_list = ["CH_BGer"]  # default to BGer
+
         self.logger.info(f"Creating {self.dataset_name} dataset")
 
         # TODO in the future: maybe save text as list of paragraphs
         # TODO make sure that the same data is saved to kaggle, csv and huggingface format!
 
-        dataset, labels = self.prepare_dataset(save_reports)
-        dataset = dataset.shuffle(seed=42)
-        self.save_dataset(dataset, labels, self.get_dataset_folder(), self.split_type,
-                          sub_datasets=sub_datasets, kaggle=kaggle, save_reports=save_reports)
+        not_created = []
+        created = []
+
+        datasets_list = []
+        labels_list = []
+
+        for court_string in court_list:
+            self.logger.info(f"Creating dataset for {court_string}")
+            dataset, labels = self.prepare_dataset(save_reports, court_string=court_string)
+
+            if len(dataset) == 0:
+                self.logger.info(f"Dataset for {court_string} could not be created")
+                not_created.append(court_string)
+            else:
+                dataset = dataset.shuffle(seed=42)
+                if concatenate:
+                    datasets_list.append(dataset)
+                    labels_list.append(labels)
+                else:  # save each dataset separately
+                    save_path = Path(os.path.join(self.get_dataset_folder(), court_string))
+                    self.save_dataset(dataset, labels, save_path, self.split_type,
+                                      sub_datasets=sub_datasets, kaggle=kaggle, save_reports=save_reports)
+
+                created.append(court_string)
+            self.logger.info(f"Empty courts: {not_created}")
+            self.logger.info(f"Created courts: {created}")
+
+        if concatenate:
+            dataset = concatenate_datasets(datasets_list)
+            labels = labels_list[0]
```

Comment:
Is this labels_list really necessary?

Reply:
If we assume that all labels are exactly the same, no. 
I kept it just in case, but if we run into a problem when concatenating datasets, we could just reintroduce it, so I'll remove it.

Reply:
Aah, yes I see. But if the labels are not the same everywhere, we probably would want the union of the labels and not just the first one, right?

Reply:
Yes, there would be additional code needed for this case.